src/openrct2-ui/windows/SavePrompt.cpp

```diff
@@ -119,133 +211,28 @@ rct_window* WindowSavePromptOpen()
-    if (stringId == STR_QUIT_GAME_PROMPT_TITLE && gScreenFlags & SCREEN_FLAGS_SCENARIO_EDITOR)
-        stringId = STR_QUIT_SCENARIO_EDITOR;
-    window_save_prompt_widgets[WIDX_TITLE].text = stringId;
-    window_save_prompt_widgets[WIDX_LABEL].text = window_save_prompt_labels[EnumValue(prompt_mode)][1];
-
-    return window;
-}
-
-/**
- *
- *  rct2: 0x0066DF17
- */
-static void WindowSavePromptClose(rct_window* w)
-{
-    // Unpause the game
-    if (network_get_mode() == NETWORK_MODE_NONE)
-    {
-        gGamePaused &= ~GAME_PAUSED_MODAL;
-        OpenRCT2::Audio::Resume();
-    }
-
-    window_invalidate_by_class(WindowClass::TopToolbar);
-}
 
-/**
- *
- *  rct2: 0x0066DDF2
- */
-static void WindowSavePromptMouseup(rct_window* w, WidgetIndex widgetIndex)
-{
-    if (gScreenFlags & (SCREEN_FLAGS_TITLE_DEMO | SCREEN_FLAGS_TRACK_DESIGNER | SCREEN_FLAGS_TRACK_MANAGER))
-    {
-        switch (widgetIndex)
-        {
-            case WQIDX_OK:
-                game_load_or_quit_no_save_prompt();
-                break;
-            case WQIDX_CLOSE:
-            case WQIDX_CANCEL:
-                window_close(*w);
-                break;
-        }
-        return;
-    }
-
-    switch (widgetIndex)
+    int32_t width = WW_SAVE;
+    int32_t height = WH_SAVE;
+    if (gScreenFlags & (SCREEN_FLAGS_TRACK_DESIGNER | SCREEN_FLAGS_TRACK_MANAGER))
     {
-        case WIDX_SAVE:
-        {
-            std::unique_ptr<Intent> intent;
-
-            if (gScreenFlags & (SCREEN_FLAGS_EDITOR))
-            {
-                intent = std::make_unique<Intent>(WindowClass::Loadsave);
-                intent->putExtra(INTENT_EXTRA_LOADSAVE_TYPE, LOADSAVETYPE_SAVE | LOADSAVETYPE_LANDSCAPE);
-                intent->putExtra(INTENT_EXTRA_PATH, gScenarioName);
-            }
-            else
-            {
-                intent = create_save_game_as_intent();
-            }
-            window_close(*w);
-            intent->putExtra(INTENT_EXTRA_CALLBACK, reinterpret_cast<void*>(WindowSavePromptCallback));
-            context_open_intent(intent.get());
-            break;
-        }
-        case WIDX_DONT_SAVE:
-            game_load_or_quit_no_save_prompt();
-            return;
-        case WIDX_CLOSE:
-        case WIDX_CANCEL:
-            window_close(*w);
-            return;
+        width = WW_QUIT;
+        height = WH_QUIT;
     }
-}
-
-static void WindowSavePromptPaint(rct_window* w, rct_drawpixelinfo* dpi)
-{
-    WindowDrawWidgets(*w, dpi);
-}
 
-static void WindowSavePromptCallback(int32_t result, const utf8* path)
-{
-    if (result == MODAL_RESULT_OK)
-    {
-        game_load_or_quit_no_save_prompt();
-    }
+    auto savePromptWindow = std::make_unique<SavePromptWindow>(prompt_mode);
+    return static_cast<SavePromptWindow*>(WindowCreate(
+        std::move(savePromptWindow), WindowClass::SavePrompt, {}, width, height,
+        WF_TRANSPARENT | WF_STICK_TO_FRONT | WF_CENTRE_SCREEN));
```

Comment:
The window is not being created centred, you probably need to add `WF_AUTO_POSITION`

```suggestion
        WF_TRANSPARENT | WF_STICK_TO_FRONT | WF_CENTRE_SCREEN | WF_AUTO_POSITION));
```